sdks/python/apache_beam/runners/worker/operations.py

```diff
@@ -710,13 +710,12 @@ def start(self):
   def process(self, o):
     # type: (WindowedValue) -> None
     with self.scoped_process_state:
-      delayed_application = self.dofn_runner.process(o)
-      if delayed_application:
+      delayed_applications = self.dofn_runner.process(o)
+      if delayed_applications:
         assert self.execution_context is not None
-        # TODO(BEAM-77746): there's disagreement between subclasses
-        #  of DoFnRunner over the return type annotations of process().
-        self.execution_context.delayed_applications.append(
-            (self, delayed_application))  # type: ignore[arg-type]
+        for delayed_application in delayed_applications:
```

Comment:
I would have expected something that is being returned here would break splitting since we would be returning appending a `Tuple[operations.DoOperation, Iterable[common.SplitResultResidual]]` instead of `Tuple[operations.DoOperation, common.SplitResultResidual]`. Maybe we are unpacking the iterable somewhere inexplicably.

We have several SDF based validates runner tests, would be worthwhile to see what they say for Dataflow.

Reply:
If the tests keep passing with the validates runner tests for SDF for a couple of runners that support portable execution (e.g. Flink/Dataflow) then we should merge this.

Reply:
The snippets I linked above are the only places I saw  `execution_context.delayed_applications` get accessed.